Userland/Libraries/LibWeb/HTML/Window.cpp

```diff
@@ -1060,6 +1064,54 @@ HTML::BrowsingContext* Window::browsing_context()
     return m_associated_document->browsing_context();
 }
 
+// https://html.spec.whatwg.org/multipage/system-state.html#pdf-viewer-plugin-objects
+Vector<JS::NonnullGCPtr<Plugin>> Window::pdf_viewer_plugin_objects()
+{
+    // Each Window object has a PDF viewer plugin objects list. If the user agent's PDF viewer supported is false, then it is the empty list.
+    // Otherwise, it is a list containing five Plugin objects, whose names are, respectively:
+    // 0.   "PDF Viewer"
+    // 1.   "Chrome PDF Viewer"
+    // 2.   "Chromium PDF Viewer"
+    // 3.   "Microsoft Edge PDF Viewer"
+    // 4.   "WebKit built-in PDF"
+    // The values of the above list form the PDF viewer plugin names list. https://html.spec.whatwg.org/multipage/system-state.html#pdf-viewer-plugin-names
+    VERIFY(page());
+    if (!page()->pdf_viewer_supported())
+        return {};
+
+    if (m_pdf_viewer_plugin_objects.is_empty()) {
+        // FIXME: Remove the MUSTs and propagate the errors instead.
+        m_pdf_viewer_plugin_objects.append(realm().heap().allocate<Plugin>(realm(), realm(), MUST(String::from_utf8("PDF Viewer"sv))).release_allocated_value_but_fixme_should_propagate_errors());
+        m_pdf_viewer_plugin_objects.append(realm().heap().allocate<Plugin>(realm(), realm(), MUST(String::from_utf8("Chrome PDF Viewer"sv))).release_allocated_value_but_fixme_should_propagate_errors());
+        m_pdf_viewer_plugin_objects.append(realm().heap().allocate<Plugin>(realm(), realm(), MUST(String::from_utf8("Chromium PDF Viewer"sv))).release_allocated_value_but_fixme_should_propagate_errors());
+        m_pdf_viewer_plugin_objects.append(realm().heap().allocate<Plugin>(realm(), realm(), MUST(String::from_utf8("Microsoft Edge PDF Viewer"sv))).release_allocated_value_but_fixme_should_propagate_errors());
+        m_pdf_viewer_plugin_objects.append(realm().heap().allocate<Plugin>(realm(), realm(), MUST(String::from_utf8("WebKit built-in PDF"sv))).release_allocated_value_but_fixme_should_propagate_errors());
```

Comment:
Don't use `MUST` for errors that should be propagated, not even with a FIXME - we have `release_allocated_value_but_fixme_should_propagate_errors` for that